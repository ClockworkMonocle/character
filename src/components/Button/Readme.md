Buttons to do a whole buncha stuff...

```js
<div>
  <span style={{ marginRight: '4px' }}>
    <Button size="sm">Small</Button>
  </span>
  <span style={{ marginRight: '4px' }}>
    <Button>Medium</Button>
  </span>
  <span style={{ marginRight: '4px' }}>
    <Button size="lg">Large</Button>
  </span>
</div>
```
```js
<div>
  <span style={{ marginRight: '4px' }}>
    <Button>Default</Button>
  </span>
  <span style={{ marginRight: '4px' }}>
    <Button variant="pill">Pill Button</Button>
  </span>
  <span style={{ marginRight: '4px' }}>
    <Button variant="bare">Bare</Button>
  </span>
</div>
```
```js
<div>
  <span style={{ marginRight: '4px' }}>
    <Button variant="default" disabled>Disabled</Button>
  </span>
  <span style={{ marginRight: '4px' }}>
    <Button variant="pill" disabled>Disabled</Button>
  </span>
  <span style={{ marginRight: '4px' }}>
    <Button variant="bare" disabled>Disabled</Button>
  </span>
</div>
```
```js
<div>
    <span style={{ marginRight: '4px' }}>
      <Button color="str">Strength</Button>
    </span>
    <span style={{ marginRight: '4px' }}>
      <Button color="dex">Dexterity</Button>
    </span>
    <span style={{ marginRight: '4px' }}>
      <Button color="con">Constitution</Button>
    </span>
    <span style={{ marginRight: '4px' }}>
      <Button color="wis">Wisdom</Button>
    </span>
    <span style={{ marginRight: '4px' }}>
      <Button color="int">Intellect</Button>
    </span>
    <span style={{ marginRight: '4px' }}>
      <Button color="cha">Charisma</Button>
    </span>
</div>
```
```js
<div>
    <span style={{ marginRight: '4px' }}>
      <Button color="proficient">Proficient</Button>
    </span>
    <span style={{ marginRight: '4px' }}>
      <Button color="attack">Attack</Button>
    </span>
    <span style={{ marginRight: '4px' }}>
      <Button color="spell">Spells</Button>
    </span>
</div>
```
