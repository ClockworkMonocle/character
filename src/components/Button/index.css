
.button {
  appearance: none;
  outline: none;
  font-size: 1em;
  border: 0;
  background-color: transparent;
  padding: 10px 30px;
  border-radius: 5px;
  cursor: pointer;
}

.button:disabled {
  opacity: 0.65;
  cursor: not-allowed;
}

.button__sm {
  font-size: 9px;
}

.button__lg {
  font-size: 1.5em;
}

.button__default {
}

.button__pill {
  border-radius: 35px;
}

.button__bare {
  padding: 0;
  background-color: transparent;
  text-transform: uppercase;
  font-weight: bold;
  letter-spacing: 2px;
}

.button__str {
  background-color: var(--str-color);
  color: var(--white);
}
.button__str:hover:not(:disabled) {
  background-color: var(--light-str-color);
}
.button__str:active:not(:disabled) {
  background-color: var(--dark-str-color);
}

.button__dex {
  background-color: var(--dex-color);
  color: var(--white);
}
.button__dex:hover:not(:disabled) {
  background-color: var(--light-dex-color);
}
.button__dex:active:not(:disabled) {
  background-color: var(--dark-dex-color);
}

.button__con {
  background-color: var(--con-color);
  color: var(--white);
}
.button__con:hover:not(:disabled) {
  background-color: var(--light-con-color);
}
.button__con:active:not(:disabled) {
  background-color: var(--dark-con-color);
}

.button__wis {
  background-color: var(--wis-color);
  color: var(--white);
}
.button__wis:hover:not(:disabled) {
  background-color: var(--light-wis-color);
}
.button__wis:active:not(:disabled) {
  background-color: var(--dark-wis-color);
}

.button__int {
  background-color: var(--int-color);
  color: var(--white);
}
.button__int:hover:not(:disabled) {
  background-color: var(--light-int-color);
}
.button__int:active:not(:disabled) {
  background-color: var(--dark-int-color);
}

.button__cha {
  background-color: var(--cha-color);
  color: var(--white);
}
.button__cha:hover:not(:disabled) {
  background-color: var(--light-cha-color);
}
.button__cha:active:not(:disabled) {
  background-color: var(--dark-cha-color);
}

.button__attack {
  background-color: var(--attack-color);
  color: var(--white);
}
.button__attack:hover:not(:disabled) {
  background-color: var(--light-attack-color);
}
.button__attack:active:not(:disabled) {
  background-color: var(--dark-attack-color);
}

.button__spell {
  background-color: var(--spell-color);
  color: var(--white);
}
.button__spell:hover:not(:disabled) {
  background-color: var(--light-spell-color);
}
.button__spell:active:not(:disabled) {
  background-color: var(--dark-spell-color);
}

.button__proficient {
  background-color: var(--dark);
  color: var(--white);
}
.button__proficient:hover:not(:disabled) {
  background-color: var(--grey-55);
}
.button__proficient:active:not(:disabled) {
  background-color: var(--grey-14);
}

.button__green {
  background-color: var(--cha-color);
  color: var(--white);
}
.button__green:hover:not(:disabled) {
  background-color: var(--light-cha-color);
}
.button__green:active:not(:disabled) {
  background-color: var(--dark-cha-color);
}

.button__red {
  background-color: var(--attack-color);
  color: var(--white);
}
.button__red:hover:not(:disabled) {
  background-color: var(--light-attack-color);
}
.button__red:active:not(:disabled) {
  background-color: var(--dark-attack-color);
}

.button__bare.button__str {
  background-color: transparent;
  color: var(--str-color);
}
.button__bare.button__str:hover:not(:disabled) {
  background-color: transparent;
  color: var(--light-str-color);
}
.button__bare.button__str:active:not(:disabled) {
  background-color: transparent;
  color: var(--dark-str-color);
}

.button__bare.button__dex {
  background-color: transparent;
  color: var(--dex-color);
}
.button__bare.button__dex:hover:not(:disabled) {
  background-color: transparent;
  color: var(--light-dex-color);
}
.button__bare.button__dex:active:not(:disabled) {
  background-color: transparent;
  color: var(--dark-dex-color);
}

.button__bare.button__con {
  background-color: transparent;
  color: var(--con-color);
}
.button__bare.button__con:hover:not(:disabled) {
  background-color: transparent;
  color: var(--light-con-color);
}
.button__bare.button__con:active:not(:disabled) {
  background-color: transparent;
  color: var(--dark-con-color);
}

.button__bare.button__wis {
  background-color: transparent;
  color: var(--wis-color);
}
.button__bare.button__wis:hover:not(:disabled) {
  background-color: transparent;
  color: var(--light-wis-color);
}
.button__bare.button__wis:active:not(:disabled) {
  background-color: transparent;
  color: var(--dark-wis-color);
}

.button__bare.button__int {
  background-color: transparent;
  color: var(--int-color);
}
.button__bare.button__int:hover:not(:disabled) {
  background-color: transparent;
  color: var(--light-int-color);
}
.button__bare.button__int:active:not(:disabled) {
  background-color: transparent;
  color: var(--dark-int-color);
}

.button__bare.button__cha {
  background-color: transparent;
  color: var(--cha-color);
}
.button__bare.button__cha:hover:not(:disabled) {
  background-color: transparent;
  color: var(--light-cha-color);
}
.button__bare.button__cha:active:not(:disabled) {
  background-color: transparent;
  color: var(--dark-cha-color);
}

.button__bare.button__attack {
  background-color: transparent;
  color: var(--attack-color);
}
.button__bare.button__attack:hover:not(:disabled) {
  background-color: transparent;
  color: var(--light-attack-color);
}
.button__bare.button__attack:active:not(:disabled) {
  background-color: transparent;
  color: var(--dark-attack-color);
}

.button__bare.button__spell {
  background-color: transparent;
  color: var(--spell-color);
}
.button__bare.button__spell:hover:not(:disabled) {
  background-color: transparent;
  color: var(--light-spell-color);
}
.button__bare.button__spell:active:not(:disabled) {
  background-color: transparent;
  color: var(--dark-spell-color);
}

.button__bare.button__proficient {
  background-color: transparent;
  color: var(--dark);
}
.button__bare.button__proficient:hover:not(:disabled) {
  background-color: transparent;
  color: var(--grey-55);
}
.button__bare.button__proficient:active:not(:disabled) {
  background-color: transpar14t;
  color: var(--grey-55);
}

.button__bare.button__green {
  background-color: transparent;
  color: var(--cha-color);
}
.button__bare.button__green:hover:not(:disabled) {
  background-color: transparent;
  color: var(--light-cha-color);
}
.button__bare.button__green:active:not(:disabled) {
  background-color: transparent;
  color: var(--dark-cha-color);
}

.button__bare.button__red {
  background-color: transparent;
  color: var(--attack-color);
}
.button__bare.button__red:hover:not(:disabled) {
  background-color: transparent;
  color: var(--light-attack-color);
}
.button__bare.button__red:active:not(:disabled) {
  background-color: transparent;
  color: var(--dark-attack-color);
}
