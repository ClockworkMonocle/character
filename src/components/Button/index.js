import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import './index.css';

export default function Button({ className, children, color, size, variant, ...props }) {
  const css = cn(
    'button',
    `button__${variant}`,
    `button__${color}`,
    `button__${size}`,
    className
  );

  return (
    <button
      className={css}
      {...props}
    >{children}</button>
  );
}

Button.propTypes = {
  variant: PropTypes.oneOf([
    'default',
    'pill',
    'bare',
  ]),
  color: PropTypes.oneOf([
    'str',
    'dex',
    'con',
    'wis',
    'int',
    'cha',
    'attack',
    'spell',
    'proficient',
    'green',
    'red',
  ]),
  size: PropTypes.oneOf([
    'sm',
    'md',
    'lg',
  ]),
};

Button.defaultProps = {
  variant: 'default',
  color: 'green',
  size: 'md',
};
