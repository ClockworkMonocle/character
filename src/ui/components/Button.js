'use strict';

var React = require('react/addons');
var cn = require('classnames');

module.exports = React.createClass({
  displayName : 'Button',


  getDefaultProps : function() {
    return ({
      size : 'sm'
    })
  },


  render : function() {
    return (
      <button>{this.props.children}</button>
    )
  }
})
