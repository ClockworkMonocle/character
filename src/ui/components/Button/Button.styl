.button
  appearance none
  outline none
  border 1px solid #ddd
  background-color $base-background-color
  border-radius $base-border-radius
  display inline-block
  margin-bottom $base-component-margin
  padding-top $base-padding-sm
  padding-bottom $base-padding-sm
  padding-left $base-padding-lg
  padding-right $base-padding-lg
  &:first-of-type
    margin-right $base-component-margin
  &:active
    background-color darken($base-background-color, 15%)
    box-shadow inset 0px 0px 3px 1px rgba(0,0,0,0.1)
  &.button-sm
    font-size 12px
  &.button-lg
    font-size 16px
  &.button-bl
    background-color $bg-blue
    color $text-blue
    border-color lighten($text-blue, 50%)
    &:active
      background-color darken($bg-blue, 15%)
      box-shadow inset 0px 0px 3px 1px rgba(0,0,0,0.1)
  &.button-gr
    background-color $bg-green
    color $text-green
    border-color lighten($text-green, 50%)
    &:active
      background-color darken($bg-green, 15%)
      box-shadow inset 0px 0px 3px 1px rgba(0,0,0,0.1)
  &.button-gd
    background-color $bg-gold
    color $text-gold
    border-color lighten($text-gold, 50%)
    &:active
      background-color darken($bg-gold, 15%)
      box-shadow inset 0px 0px 3px 1px rgba(0,0,0,0.1)
  &.button-rd
    background-color $bg-red
    color $text-red
    border-color lighten($text-red, 50%)
    &:active
      background-color darken($bg-red, 15%)
      box-shadow inset 0px 0px 3px 1px rgba(0,0,0,0.1)
