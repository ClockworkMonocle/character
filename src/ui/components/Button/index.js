'use strict';

import React from 'react';
import cn from 'classnames';

module.exports = React.createClass({
  displayName : 'Button',


  getDefaultProps : function() {
    return ({
      size : 'md'
    })
  },


  render : function() {
    let { size, style, children, ...props } = this.props;
    let css = cn({
      'button' : true,
      'button-sm' : size === 'sm',
      'button-md' : size === 'md',
      'button-lg' : size === 'lg',
      'button-sq' : size === 'sq',
      'button-bl' : style === 'blue',
      'button-gr' : style === 'green',
      'button-rd' : style === 'red',
      'button-gd' : style === 'gold'
    })

    return (
      <button className={css} {...props}>{children}</button>
    )
  }
})
