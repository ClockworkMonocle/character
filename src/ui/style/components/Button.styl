@require('base.styl')

.button
  appearance none
  outline none
  border 1px solid #ddd
  background-color #fff
  border-radius $base-border-radius
  display inline-block
  margin-right $base-component-margin
  margin-bottom $base-component-margin
  &:active
    background-color darken(#fff, 15%)
  &.button-sm
    font-size 12px
  &.button-lg
    font-size 16px
  &.button-bl
    background-color $bg-blue
    color $text-blue
    border-color lighten($text-blue, 50%)
    &:active
      background-color darken($bg-blue, 15%)
      box-shadow inset 0px 0px 3px 1px rgba(0,0,0,0.1)
  &.button-gr
    background-color $bg-green
    color $text-green
    border-color lighten($text-green, 50%)
    &:active
      background-color darken($bg-green, 15%)
      box-shadow inset 0px 0px 3px 1px rgba(0,0,0,0.1)
  &.button-gd
    background-color $bg-gold
    color $text-gold
    border-color lighten($text-gold, 50%)
    &:active
      background-color darken($bg-gold, 15%)
      box-shadow inset 0px 0px 3px 1px rgba(0,0,0,0.1)
  &.button-rd
    background-color $bg-red
    color $text-red
    border-color lighten($text-red, 50%)
    &:active
      background-color darken($bg-red, 15%)
      box-shadow inset 0px 0px 3px 1px rgba(0,0,0,0.1)
