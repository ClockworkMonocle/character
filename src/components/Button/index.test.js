import React from 'react';
import { shallow } from 'enzyme';
import Button from './index';

test('<Button /> renders defaults', () => {
  expect(shallow(<Button>heyo</Button>)).toMatchSnapshot();
});

test('<Button /> renders size sm', () => {
  expect(shallow(<Button size="sm">small</Button>)).toMatchSnapshot();
});

test('<Button /> renders size lg', () => {
  expect(shallow(<Button size="lg">large</Button>)).toMatchSnapshot();
});

test('<Button /> renders pill', () => {
  expect(shallow(<Button variant="pill">Pill</Button>)).toMatchSnapshot();
});

test('<Button /> renders bare', () => {
  expect(shallow(<Button variant="bare">Bare</Button>)).toMatchSnapshot();
});
