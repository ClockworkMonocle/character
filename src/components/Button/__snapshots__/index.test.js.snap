// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders bare 1`] = `
<button
  className="button button__bare button__green button__md"
>
  Bare
</button>
`;

exports[`<Button /> renders defaults 1`] = `
<button
  className="button button__default button__green button__md"
>
  heyo
</button>
`;

exports[`<Button /> renders pill 1`] = `
<button
  className="button button__pill button__green button__md"
>
  Pill
</button>
`;

exports[`<Button /> renders size lg 1`] = `
<button
  className="button button__default button__green button__lg"
>
  large
</button>
`;

exports[`<Button /> renders size sm 1`] = `
<button
  className="button button__default button__green button__sm"
>
  small
</button>
`;
